CHANGELOG.md

```diff
@@ -1,3 +1,28 @@
+# 2.0.0
+
+## What's Changed
+
+### Breaking changes
+
+* Remove `TxGroup.assignGroupID(Transaction[] txns, Address address)` in favor
+  of `TxGroup.assignGroupID(Address address, Transaction ...txns)`.
+* Remove `Account.transactionWithSuggestedFeePerByte` in favor of `Account.setFeeByFeePerByte`.
+* Remove deprecated methods in `Transaction.java`, mark public `Transaction` constructor as hidden in favor of `com.algorand.algosdk.builder.transaction`.
+* Remove deprecated `Transaction.setFee` and `Transaction.setLease` methods.
+* Remove v1 algod API (`com.algorand.algosdk.algod.client`) due to API end-of-life (2022-12-01). Instead, use v2 algod API (`com.algorand.algosdk.v2.client.algod`).
+* Remove `cost` field in `DryrunTxnResult` in favor of 2 fields:  `budget-added` and `budget-consumed`. `cost` can be derived by `budget-consumed - budget-added`.
+* Remove logicsig templates, `com/algorand/algosdk/resource/langspec.json`, `com.algorand.algosdk.logic` and methods in `com.algorand.algosdk.crypto` depending on `langspec.json`.
+* Remove the deprecated `MethodCallParams` public constructor in favor of `com.algorand.algosdk.builder.transaction.MethodCallTransactionBuilder`.
+* Remove unused generated types:  `CatchpointAbortResponse`, `CatchpointStartResponse`.
+
+### Enhancements
+
+* Enhancement: Sourcemap `mapping` field deprecation by @ahangsu in https://github.com/algorand/java-algorand-sdk/pull/441
+* Enhancement: Remove v1 algod endpoint by @ahangsu in https://github.com/algorand/java-algorand-sdk/pull/443
+* v2: V1 to V2 breaking changes by @ahangsu in https://github.com/algorand/java-algorand-sdk/pull/447
```

Comment:
Optional - For this release, I think the already listed breaking changes account for the included changes.

```suggestion
```